lib/beacon/multicast.go

```diff
@@ -152,33 +95,23 @@ func (w *multicastWriter) serve(stop chan struct{}) error {
 			}
 		}
 
-		if success > 0 {
-			w.SetError(nil)
+		if success == 0 {
+			return err
 		}
 	}
 }
 
-func (w *multicastWriter) String() string {
-	return fmt.Sprintf("multicastWriter@%p", w)
-}
+func readMulticasts(outbox chan<- recv, addr string, stop chan struct{}) error {
+	l.Debugln("starting readMulticasts")
+	defer l.Debugln("stopping readMulticasts")
 
```

Comment:
The cast could log this cruft I guess? It has the name

Reply:
Not really, as it's just the supervisor, it doesn't invoke these methods itself.